### PR TITLE
Minor item changes

### PR DESCRIPTION
**Pull request by Thundermaker300**
- Added properties to Item class to identify what type of item it is (without casting)
- Added return types to a few methods in Scp330 (these methods previously returned void, so this shouldn't be breaking)
  - AddCandy now returns whether or not the candy was successfully added
  - RemoveCandy now returns the amount of candies that were removed
- Added return type to `Player.AddItem(ItemType itemType, int amount)`. and `Player.AddItem(List<ItemType> items)`
- Added `Ammo.AmmoLimit` - Constant set to `65535`, which is what the game will cap out ammo count

### CODE REVIEW

**Review by joker-119 — CHANGES_REQUESTED** [review]
Changing a void method to have a return type is a breaking change because the method signature changes. These need to be made as overloads with out parameters, with the old methods marked obsolete, then when we version bump to 5.x we can remove the out parameters and make them return values instead like we did for custom items between 3.x and 4.x

**Comment by Thundermaker300** [comment]
Understandable, I'll make the changes when I get on my computer later today

**Review by joker-119 — CHANGES_REQUESTED** [review]
Include the true value in the obsolete attributes to enforce build errors.

`[Obsolete("Use xyz", true]`

**Comment by Thundermaker300** [comment]
omg true!!!!!!!!!!!

File: Exiled.API/Features/Items/Ammo.cs [hunk 1]
```diff
@@ -16,6 +16,11 @@ namespace Exiled.API.Features.Items
     /// </summary>
     public class Ammo : Item
     {
+        /// <summary>
+        /// Specifies the maximum amount of ammo that may be held at one time.
+        /// </summary>
+        public const ushort AmmoLimit = 65535;
```
Comment (joker-119) [inline Exiled.API/Features/Items/Ammo.cs:22]: This is a breaking change.

**Review by joker-119 — CHANGES_REQUESTED** [review]

File: Exiled.API/Features/Items/Ammo.cs (hunk 1 above)
Comment (Thundermaker300) [reply Exiled.API/Features/Items/Ammo.cs:22]: How so? It's a new property that didn't exist before.

Comment (joker-119) [reply Exiled.API/Features/Items/Ammo.cs:22]: It's not, I'm retarded

**Review by joker-119 — APPROVED** [review]

**Review by joker-119 — CHANGES_REQUESTED** [review]
Fix build errors

**Comment by joker-119** [comment]
re-merge dev into this branch :)

**Review by joker-119 — APPROVED** [review]

**Review by BuildBoy12 — APPROVED** [review]

File: Exiled.API/Features/Items/Scp330.cs [hunk 2]
```diff
@@ -65,22 +65,53 @@ public Scp330()
         /// Adds a specific candy to the bag.
         /// </summary>
         /// <param name="type">The <see cref="CandyKindID"/> to add.</param>
+        [System.Obsolete("Please use RemoveCandy(CandyKindID, out bool) instead.", true)]
```
Comment (Jesus-QC) [inline Exiled.API/Features/Items/Scp330.cs:68]: Please use AddCandy

**Review by Jesus-QC — CHANGES_REQUESTED** [review]

File: Exiled.API/Features/Items/Ammo.cs [hunk 3]
```diff
@@ -16,6 +16,14 @@ namespace Exiled.API.Features.Items
     /// </summary>
     public class Ammo : Item
     {
+        /// <summary>
+        /// Specifies the absolute maximum amount of ammo that may be held at one time, if ammo is forcefully given to the player (regardless of worn armor or server configuration).
+        /// <para>
+        /// For accessing the maximum amount of ammo that may be held based on worn armor and server settings, see <see cref="Player.GetAmmoLimit(ItemType)"/>.
+        /// </para>
+        /// </summary>
+        public const ushort AmmoLimit = 65535;
```
Comment (Jesus-QC) [inline Exiled.API/Features/Items/Ammo.cs:25]: what do you think about 
=> ushort.MaxValue

**Review by Jesus-QC — CHANGES_REQUESTED** [review]

File: Exiled.API/Features/Items/Ammo.cs (hunk 3 above)
Comment (joker-119) [reply Exiled.API/Features/Items/Ammo.cs:25]: I agree this should be a property

File: Exiled.API/Features/Items/Scp330.cs (hunk 2 above)
Comment (Thundermaker300) [reply Exiled.API/Features/Items/Scp330.cs:68]: good catch

**Pull request merged**